rex/resource.py

```diff
@@ -1253,11 +1254,130 @@ def _get_ds(self, ds_name, ds_slice):
+            each arg is for a sequential axis
+
+        Returns
+        -------
+        out : ndarray
+            ndarray of variable timeseries data
+            If unscale, returned in native units else in scaled units
+        """
+        ti_shape = self.shapes.get('time_index')
+        meta_shape = self.shapes.get('meta')
+        if ti_shape == meta_shape:
+            msg = ("Attempting to use a 2D slice on a 1D dataset when the "
+                   "meta and time index have the same shape - unable to "
+                   "disambiguate the slice dimensions. Please either update "
+                   "the length of the meta and/or index, set the shape of "
+                   "{!r} to be 2-dimensional (current shape: {!r}), or use a "
+                   "1-dimensional slice.".format(ds_name, ds.shape))
+            raise ResourceRuntimeError(msg)
+
+        if ds.shape == ti_shape:
+            return self._get_ds_with_spatial_repeat(ds, ds_name, ds_slice)
+        if ds.shape == meta_shape:
+            return self._get_ds_with_temporal_repeat(ds, ds_name, ds_slice)
+
+        msg = ("Attempting to use a 2D slice on a 1D dataset ({0!r}) when "
+               "the shape of the dataset {1!r} does not match the shape "
+               "of the meta {2!r})or the time index {3!r}. Please either "
+               "update the length of ({0!r}) to match either the meta or "
+               "index, or use a 1-dimensional slice."
+               .format(ds_name, ds.shape, meta_shape, ti_shape))
+        raise ResourceRuntimeError(msg)
+
+    def _get_ds_with_spatial_repeat(self, ds, ds_name, ds_slice):
+        """
+        Extract 1D data using 2D slice by repeating the 1D data along
+        the spatial dimension
+
+        Parameters
+        ----------
+        ds : h5py.dataset
+            Open .h5 dataset instance to extract data from
+        ds_name : str
+            Variable dataset to be extracted
+        ds_slice : tuple
+            Tuple of (int, slice, list, ndarray) of what to extract from ds,
+            each arg is for a sequential axis
+
+        Returns
+        -------
+        out : ndarray
+            ndarray of variable timeseries data
+            If unscale, returned in native units else in scaled units
+        """
+        msg = ("Attempting to use a 2D slice on a 1D dataset ({!r}). 1D "
+               "dataset values will be repeated along the spatial dimension"
+               .format(ds_name))
+        warn(msg)
+
+        out = ResourceDataset.extract(ds, ds_slice[:1],
+                                      scale_attr=self.SCALE_ATTR,
                                       add_attr=self.ADD_ATTR,
                                       unscale=self._unscale)
+        try:
+            out = (np.ones((len(out), self.shapes['meta'][0])).T * out).T
+            return out[:, ds_slice[1]]
+        except TypeError:
+            out = np.ones(self.shapes['meta']) * out
+            return out[ds_slice[1]]
```

Comment:
What typeerror are you catching here? Seems funky.

Reply:
If "out" is an int or float value, we cannot get it's length. In the updated version of the code, a TypeError is thrown if an int or float is attempted to be subscripted. I will add a clarifying comment by the TypeError

Reply:
Gotcha, can you do this instead of the try except? cleaner and more explicit:
```
if not isinstance(out, np.ndarray):
    out *= np.ones(shape, dtype=np.float32)

Reply:
sure